src/NuGet.Clients/NuGet.Tools/NuGetPackage.cs

```diff
@@ -85,6 +85,8 @@ public sealed partial class NuGetPackage : AsyncPackage, IVsPackageExtensionProv
         private const string F1KeywordValuePmUI = "VS.NuGet.PackageManager.UI";
 
         private AsyncLazy<IVsMonitorSelection> _vsMonitorSelection;
+        private CancellationToken _cancellationToken;
```

Comment:
cancellation tokens happen on actions, so it doesn't seem right to keep them as state. Especially on the `NuGetPackage` state. I see it being set only in an initialize, which means it probably never gets cancellation requested (unless VS is closed before startup completes).

We already use `VsShellUtilities.ShutdownToken` elsewhere, so that's a better candidate to use whereever this field is currently being used.